storage/log_iterator.go

```diff
@@ -0,0 +1,655 @@
+package storage
+
+import (
+	"bufio"
+	"container/heap"
+	"context"
+	"fmt"
+	"io"
+	"runtime"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/evergreen-ci/logkeeper"
```

Comment:
I pulled everything out into a "models" package, but I'm happy to drop that in favor of your work since all I did was copy stuff over to the "models" folder...

Reply:
Also happy to get on a call and pair on this if it would be easier to hash out the design of the packages that way!